src/FieldWarning/Assets/Ingame/UI/InputManager.cs

```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
-        }
+        private ClickManager _rightClickManager;
 
-        private Vector3 _boxSelectStart;
+        private SelectionManager _selectionManager;
 
-        void Awake()
-        {
-            _selectionManager = new SelectionManager();
-            _selectionManager.Awake();
-        }
+        private MatchSession _session;
 
-        void Start()
+        public enum Events
         {
-            _firePosReticle = (Texture2D)Resources.Load("FirePosTestTexture");
-            if (_firePosReticle == null)
-                throw new Exception("No fire pos reticle specified!");
-
-            _primedReticle = (Texture2D)Resources.Load("PrimedCursor");
-            if (_primedReticle == null)
-                throw new Exception("No primed reticle specified!");
-
-            _rightClickManager = new ClickManager(1, MoveGhostsToMouse, OnOrderShortClick, OnOrderLongClick, OnOrderHold);
+            Clicked,
+            Down,
+            Released
         }
 
-        void Update()
+        public enum Modes
         {
-            _selectionManager.Update(CurMouseMode);
-
-            switch (CurMouseMode) {
-
-            case MouseMode.purchasing:
-
-                RaycastHit hit;
-                if (Util.GetTerrainClickLocation(out hit)
-                    && hit.transform.gameObject.name.Equals("Terrain")) {
-
-                    ShowGhostUnitsAndMaybePurchase(hit);
-                }
-
-                MaybeExitPurchasingMode();
-                break;
-
-            case MouseMode.normal:
-                ApplyHotkeys();
-                _rightClickManager.Update();
-                break;
-
-            case MouseMode.firePos:
-                ApplyHotkeys();
-
-                if (Input.GetMouseButtonDown(0))
-                    _selectionManager.DispatchFirePosCommand();
-
-                if ((Input.GetMouseButtonDown(0) && !Input.GetKey(KeyCode.LeftShift))
-                    || Input.GetMouseButtonDown(1))
-                    EnterNormalMode();
-
-                break;
-
-            case MouseMode.reverseMove:
-                ApplyHotkeys();
-                if (Input.GetMouseButtonDown(0)) {
-                    MoveGhostsToMouse();
-                    _selectionManager.DispatchMoveCommand(
-                            false, MoveWaypoint.MoveMode.reverseMove);
-                }
-
-                if ((Input.GetMouseButtonDown(0) && !Input.GetKey(KeyCode.LeftShift))
-                    || Input.GetMouseButtonDown(1))
-                    EnterNormalMode();
-                break;
-
-            case MouseMode.fastMove:
-                ApplyHotkeys();
-                if (Input.GetMouseButtonDown(0)) {
-                    MoveGhostsToMouse();
-                    _selectionManager.DispatchMoveCommand(
-                            false, MoveWaypoint.MoveMode.fastMove);
-                }
+            Normal,
+            Purchasing,
+            FirePosition,
+            ReverseMove,
+            FastMove
```

Comment:
Enum values should be UPPERCASE as any constant. Not that the original code was correct, but since you're renaming them anyway..